src/main/java/org/spongepowered/common/mixin/core/world/MixinWorldServer.java

```diff
@@ -1891,16 +1890,20 @@ public IBlockState getBlockState(BlockPos pos) {
             // Sponge end
             return Blocks.AIR.getDefaultState();
         } else {
+            net.minecraft.world.chunk.Chunk chunk;
             // ExtraUtilities 2 expects to get the proper chunk while mining or it gets stuck in infinite loop
-            // TODO add TE config to disable/enable chunk loads
-            final boolean forceChunkRequests = this.mixinChunkProviderServer.getForceChunkRequests();
-            final PhaseTracker phaseTracker = PhaseTracker.getInstance();
-            final IPhaseState<?> currentState = phaseTracker.getCurrentState();
-            if (currentState == TickPhase.Tick.TILE_ENTITY) {
-                ((IMixinChunkProviderServer) this.getChunkProvider()).setForceChunkRequests(true);
+            if (!this.activeConfig.getConfig().getWorld().getDenyTEChunkRequests()) {
```

Comment:
As mentioned before, you'd want to check on the `TileEntityType` instead of a global flag. Global flags are something we've moved away from for configs for things like this.

Reply:
Yep, I've just commited other local changes I was testing. I will add this in a few commits.